Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -78,45 +92,49 @@ void preparePathForEditing() {
             // Open a new path that the contents of the old path will be streamed into. We cannot re-use the same path as it is now closed
             // and cannot be opened again. We use the newPath variable because the factory was returning the same pointer for some strange
             // reason so this will force it to do otherwise.
-            FAIL_FAST_IF_FAILED(factory->CreatePathGeometry(&newPath));
-            FAIL_FAST_IF_FAILED(newPath->Open(&newSink));
-            FAIL_FAST_IF_FAILED(_impl.pathGeometry->Stream(newSink.Get()));
+            RETURN_IF_FAILED(factory->CreatePathGeometry(&newPath));
+            RETURN_IF_FAILED(newPath->Open(&newSink));
+            RETURN_IF_FAILED(PathGeometry()->Stream(newSink.Get()));
 
-            _impl.pathGeometry = newPath;
-            _impl.geometrySink = newSink;
+            PathGeometry() = newPath;
+            GeometrySink() = newSink;
 
             // Without a new figure being created, it's by default closed
-            _impl.isFigureClosed = true;
+            _impl.figureClosed = true;
         }
+        return S_OK;
     }
 
-    void closePath() {
-        if (_impl.geometrySink) {
-            endFigure(D2D1_FIGURE_END_OPEN);
-            _impl.geometrySink->Close();
-            _impl.geometrySink = nullptr;
+    void ClosePath() {
+        if (GeometrySink()) {
```

Comment:
The right way to keep _impl encapsulated and still do this would be to do _impl.ClosePath() here and implement the rest of the code in the impl class.  You haven't really helped encapsulation by doing what you have now - just the variable name.